storage/CMakeLists.txt

```diff
@@ -56,16 +56,22 @@ google_cloud_cpp_add_common_options(storage_common_options)
 # Enable unit tests
 enable_testing()
 
-# Search for libcurl
+# Search for libcurl, in CMake 3.5 this does not define a target, but it does in later versions.
+# Define the target ourselves if it is missing.
 find_package(CURL REQUIRED)
 if (NOT TARGET CURL::CURL)
     add_library(CURL::CURL UNKNOWN IMPORTED)
     set_target_properties(CURL::CURL PROPERTIES INTERFACE_INCLUDE_DIRECTORIES "${CURL_INCLUDE_DIR}")
     set_property(TARGET CURL::CURL APPEND PROPERTY IMPORTED_LOCATION "${CURL_LIBRARY}")
 endif ()
 
+# Find OpenSSL and ZLIB, these define targets as-of CMake 3.5 and higher.
+find_package(OpenSSL REQUIRED)
+find_package(ZLIB REQUIRED)
```

Comment:
Looks like this is only required for Windows -- could you add a comment to that effect?

Reply:
I think (but I might be wrong), that these are actually required when linking statically, it just happens that Windows in the only platform where we are forcing static linking of system-provided libraries.  I added a comment saying that.

As to why Windows is linked statically: because grpc does not support shared libraries on Windows.  And as to why not link statically in one of the Linux builds, that is a bug, created #510 to track it.